packages/twenty-server/src/engine/core-modules/calendar/timeline-calendar-event.service.ts

```diff
@@ -1,272 +1,159 @@
 import { Injectable } from '@nestjs/common';
 
-import groupBy from 'lodash.groupby';
+import { Any } from 'typeorm';
+import omit from 'lodash.omit';
 
 import { TIMELINE_CALENDAR_EVENTS_DEFAULT_PAGE_SIZE } from 'src/engine/core-modules/calendar/constants/calendar.constants';
-import { TimelineCalendarEventParticipant } from 'src/engine/core-modules/calendar/dtos/timeline-calendar-event-participant.dto';
-import {
-  TimelineCalendarEvent,
-  TimelineCalendarEventVisibility,
-} from 'src/engine/core-modules/calendar/dtos/timeline-calendar-event.dto';
+import { TimelineCalendarEventVisibility } from 'src/engine/core-modules/calendar/dtos/timeline-calendar-event.dto';
 import { TimelineCalendarEventsWithTotal } from 'src/engine/core-modules/calendar/dtos/timeline-calendar-events-with-total.dto';
-import { WorkspaceDataSourceService } from 'src/engine/workspace-datasource/workspace-datasource.service';
-import { ObjectRecord } from 'src/engine/workspace-manager/workspace-sync-metadata/types/object-record';
-import { CalendarEventParticipantObjectMetadata } from 'src/modules/calendar/standard-objects/calendar-event-participant.object-metadata';
+import { InjectWorkspaceRepository } from 'src/engine/twenty-orm/decorators/inject-workspace-repository.decorator';
+import { WorkspaceRepository } from 'src/engine/twenty-orm/repository/workspace.repository';
+import { CalendarEventWorkspaceEntity } from 'src/modules/calendar/standard-objects/calendar-event.workspace-entity';
+import { PersonWorkspaceEntity } from 'src/modules/person/standard-objects/person.workspace-entity';
 
-type TimelineCalendarEventParticipantWithPersonInformation =
-  ObjectRecord<CalendarEventParticipantObjectMetadata> & {
-    personFirstName: string;
-    personLastName: string;
-    personAvatarUrl: string;
-    workspaceMemberFirstName: string;
-    workspaceMemberLastName: string;
-    workspaceMemberAvatarUrl: string;
-  };
 @Injectable()
 export class TimelineCalendarEventService {
   constructor(
-    private readonly workspaceDataSourceService: WorkspaceDataSourceService,
+    @InjectWorkspaceRepository(CalendarEventWorkspaceEntity)
+    private readonly calendarEventRepository: WorkspaceRepository<CalendarEventWorkspaceEntity>,
+    @InjectWorkspaceRepository(PersonWorkspaceEntity)
+    private readonly personRepository: WorkspaceRepository<PersonWorkspaceEntity>,
   ) {}
 
+  // TODO: Align return type with the entities to avoid mapping
   async getCalendarEventsFromPersonIds(
-    workspaceMemberId: string,
-    workspaceId: string,
     personIds: string[],
     page = 1,
     pageSize: number = TIMELINE_CALENDAR_EVENTS_DEFAULT_PAGE_SIZE,
   ): Promise<TimelineCalendarEventsWithTotal> {
     const offset = (page - 1) * pageSize;
 
-    const dataSourceSchema =
-      this.workspaceDataSourceService.getSchemaName(workspaceId);
+    const calendarEventIds = await this.calendarEventRepository.find({
```

Comment:
We need to do two separate calls because TypeORM by default is filtering the where condition in the relations.